lib/ansible/module_utils/tacp_ansible/tacp_utils.py

```diff
@@ -179,64 +182,215 @@ def power_action_on_instance_by_uuid(self, uuid, power_action):
             Action.ABSENT: self.api.delete_application_using_delete,
             Action.RESUMED: self.api.resume_application_using_put
         }
-        return power_action_dict[power_action](uuid)
-
-
-class ApplicationGroupResource(Resource):
-    resource_class = tacp.ApplicationGroupsApi
-
-    def filter(self, **filters):
-        return self.api.get_application_group_list_using_get(
-            **self.get_filters_kws(**filters)
-        )
-
-    @wait_to_complete
-    def create(self, name, datacenter_uuid):
-        return self.api.create_application_group_using_post(
-            tacp.ApiCreateApplicationGroupPayload(
-                name=name, datacenter_uuid=datacenter_uuid
-            )
-        )
+        return power_action_dict[power_action](uuid).to_dict()
```

Comment:
You would want to add validation here to make the method more resilient to incorrect inputs. For instance, if power_action does not exist in your mapping, throw a certain exception.